primitives/src/vrf/blsvrf.rs

```diff
@@ -98,14 +127,33 @@ where
 #[cfg(test)]
 mod test {
     use super::*;
-    use crate::vrf::tests::{failed_verification, sign_and_verify};
-    use sha2::Sha256;
+    use ark_std::{test_rng, vec};
+
+    pub(crate) fn sign_and_verify(
+        vrf: &mut BLSVRFScheme,
+        message: &<BLSVRFScheme as Vrf>::Input,
+        bad_message: &<BLSVRFScheme as Vrf>::Input,
+    ) {
+        let rng = &mut test_rng();
+
+        let parameters = vrf.param_gen(Some(rng)).unwrap();
+        let (sk, pk) = vrf.key_gen(&parameters, rng).unwrap();
+        let vrf_proof = vrf.prove(&parameters, &sk, message, rng).unwrap();
+        let _vrf_output = vrf.evaluate(&parameters, &vrf_proof).unwrap();
+        assert!(vrf.verify(&parameters, &vrf_proof, &pk, message).unwrap());
+        assert!(!vrf
+            .verify(&parameters, &vrf_proof, &pk, bad_message)
+            .unwrap());
+    }
 
     #[test]
     fn test_bls_vrf() {
-        let message = [0u8; 32];
-        let message_bad = [1u8; 32];
-        sign_and_verify::<BLSVRFScheme, Sha256>(&message);
-        failed_verification::<BLSVRFScheme, Sha256>(&message, &message_bad);
+        let message = vec![0u8; 32];
```

Comment:
Should we also test the correctness of the hash output somewhere?

Reply:
Good point, will add

Reply:
Done in https://github.com/EspressoSystems/jellyfish/pull/137/commits/b9f3c50802e78f674cef8de80722e871c23b4cff.